support/cas-server-support-oidc/src/test/java/org/apereo/cas/oidc/claims/OidcCustomScopeAttributeReleasePolicyTests.java

```diff
@@ -1,13 +1,41 @@
 package org.apereo.cas.oidc.claims;
 
+import org.apereo.cas.authentication.CoreAuthenticationTestUtils;
+import org.apereo.cas.oidc.AbstractOidcTests;
+import org.apereo.cas.oidc.OidcConstants;
+import org.apereo.cas.util.CollectionUtils;
+
+import lombok.val;
 import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /**
  * This is {@link OidcCustomScopeAttributeReleasePolicyTests}.
  *
  * @author Misagh Moayyed
  * @since 6.1.0
  */
+
+@TestPropertySource(
+    properties = "cas.authn.oidc.claims=sub,name,given_name,family_name,middle_name,preferred_username,email,mail,groups")
 @Tag("OIDC")
-public class OidcCustomScopeAttributeReleasePolicyTests {
+public class OidcCustomScopeAttributeReleasePolicyTests extends AbstractOidcTests {
+    @Test
+    public void verifyOperation() {
+        val policy = new OidcCustomScopeAttributeReleasePolicy("groups", CollectionUtils.wrap("groups"));
+        assertEquals(OidcConstants.CUSTOM_SCOPE_TYPE, policy.getScopeType());
+        assertNotNull(policy.getAllowedAttributes());
+        val principal = CoreAuthenticationTestUtils.getPrincipal(CollectionUtils.wrap("groups", List.of("admin", "user")));
+        val attrs = policy.getAttributes(principal,
+            CoreAuthenticationTestUtils.getService(),
+            CoreAuthenticationTestUtils.getRegisteredService());
+        assertTrue(policy.getAllowedAttributes().stream().allMatch(attrs::containsKey));
```

Comment:
Then, you should also do this again:

```java
val attrs = policy.getAttributes(principal,
            CoreAuthenticationTestUtils.getService(),
            CoreAuthenticationTestUtils.getRegisteredService());
```

...and make sure `attrs` has the right attributes and entries. The `principal` that is passed again should be reconstructed based on the attributes from the first call:

```java
val principal2 = CoreAuthenticationTestUtils.getPrincipal(attrs);
val releaseAttrs = policy.getAttributes(principal2,
            CoreAuthenticationTestUtils.getService(),
            CoreAuthenticationTestUtils.getRegisteredService());
```

Reply:
I tried to make these changes. I am not sure what you are referring to in the comment below about the "feed".

Reply:
Sorry, that was a stupid typo. I meant "feedback" 😁